### ISSUE TITLE
Incompatible with snapcraft v7

### ISSUE THREAD

**Issue by kyrofa** [nextcloud-snap/nextcloud-snap#2122]
Snapcraft v7 ripped out APIs we are using in our plugins (no deprecation notices or anything). As a result, the Nextcloud snap doesn't build with snapcraft v7. See [here](https://forum.snapcraft.io/t/snapcraft-7-refuses-to-build-core18-snap/30592?u=kyrofa) for more information.

We need to rewrite our plugins to work with snapcraft v7.

**Comment by kroko** [comment]
While trying to build nextcloud-snap hit by failure

```
Pulling PHP extensions...
module 'snapcraft' has no attribute 'sources'
```

which refers to https://github.com/nextcloud-snap/nextcloud-snap/blob/master/snap/plugins/php.py#L100

Using `snapcraft 7.3.1`

**Comment by kyrofa** [comment]
Yep, that's it indeed.

**Comment by pachulo** [comment]
This is the snapcraft bug report https://github.com/snapcore/snapcraft/issues/4285

**Comment by pachulo** [comment]
OK, this had been solved and the fix will be in snapcraft 7.5.3! 😄
